### ISSUE TITLE
False positive - E1101: Module 'crypt' has no 'METHOD_SHA512' member

### ISSUE THREAD

**Issue by fredj** [PyCQA/pylint#3047]
### Steps to reproduce
Run pylint on the following code:
```python
import crypt
print(crypt.crypt('hello world', crypt.METHOD_SHA512))
```
Note: `METHOD_SHA512` was added in python 3.3

### Current behavior
```shell
pylint --errors-only foo.py 
************* Module foo
/foo.py:2:33: E1101: Module 'crypt' has no 'METHOD_SHA512' member (no-member)
```
### Expected behavior
pylint should not raise an error here

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.4 (default, Jul 13 2019, 14:04:11) 
```

It's probably related to the python version because if the same code is checked on python 3.6 no error is reported:
```
pylint 2.3.1
astroid 2.2.5
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
```

**Comment by gyermolenko** [comment]
I checked the snippet against python 3.7.4 and 3.6.8 and got same results.
3.6.8 has the list of methods at module level like 
```python3
#  available salting/crypto methods
...
METHOD_SHA512 = _Method('SHA512', '6', 16, 106)
...
```

3.7.4 does it dynamically, populating module globals on import
```python3
#  available salting/crypto methods
methods = []

def _add_method(name, *args, rounds=None):
    method = _Method(name, *args)
    globals()['METHOD_' + name] = method
    ...

_add_method('SHA512', '6', 16, 106)
```

**Comment by fredj** [comment]
nice find @gyermolenko

**Comment by PCManticore** [comment]
This most likely needs an astroid patch in here: https://github.com/PyCQA/astroid/tree/master/astroid/brain

**Comment by gyermolenko** [comment]
Looks interesting! 
I can try to work on that but need to read up about astroid first
Is [this](http://pylint.pycqa.org/projects/astroid/en/latest/extending.html) a good place to start?

**Comment by PCManticore** [comment]
@gyermolenko Yeah, that looks like a good resource! Do let me know if you encounter any issues.
